### PR TITLE
Documents Model, API, db-API, Tests and using keycloak permissions as key-value parameters

### PR DESCRIPTION
**Pull request by guillecro**
### Added document model

```js
const Document = new mongoose.Schema({
  authorId: { type: String, required: true }, // Keycloak Id
  published: { type: Boolean, required: true },
  documentType: { type: mongoose.Schema.Types.ObjectId, ref: 'DocumentType' },
  documentTypeVersion: { type: Number, required: true },
  content: {
    title: { type: String, required: true, maxlength: 120 },
    brief: { type: String, required: false },
    fields: { type: mongoose.Schema.Types.Mixed }
  }
}, {
  timestamps: true,
  versionKey: false
})
```
### Created the db-api Methods
- `document.create(document, documentType)` - Before saving, data is validated with the JSON Schema defined in the documentType 
- `document.get(query)`
- `document.list(query, { limit, page })` - Paginated
- `document.update(id)` - First retrieves the document. If it exists, we merge the changes and then we validate the data with the JSON Schema defined in the documentType 
- `document.remove(id)`

### Created the API Methods
- `GET /api/v1/documents` - (Only members of the group:Accountable) Only retrieves documents where `{ authorId: blablaTheAuthorIdblabla }`
- `POST /api/v1/documents` - (Only members of the group:Accountable) - Checks how many docs the user created before and checks with the limit set for the members of group Accountable. If everything is OK, then the document gets created
- `GET /api/v1/documents/:id` - If the request comes from the author itself, the document is retrieved no matter if it is a draft or not. If not, only published documents are retrieved.
- `PUT /api/v1/documents/:id` - (Only members of the group:Accountable) - Checks the user and the author and if everything is ok, then it updates the document.
- `DELETE /api/v1/documents/:id` - (Only members of the group:Accountable) - Checks the user and the author and if they match, then it deletes the document

### Using Groups to define the set of permissions using key-value attributes
- Added new key to the user mapper for the client 'app'. It retrieves a *User attribute*, that actually is an attribute of the created Group "Accountable", which is `documentLimit`.
- The group *Accountable* was created. Members will have the realm role "accountable" and the attribute `documentLimit = 1`
- Updated realm-dev and realm-test. When creating the users, you have to assign the group to the user (not done via CLI, at least, for now).

Generated Token
```
{
  (...)
  "realm_access": {
    "roles": [
      "accountable"
    ]
  },
  "permissions": {
    "documentLimit": 1
  },
  "groups": [
    "/Accountable"
  ],
(...)
}
```

### Tests for document:
```
Community API (/api/v1/community)
    As anonymous user
      ✓ GET (/) Anyone should be able to get the community data (87ms)
    As an admin user
      ✓ should get access_token from auth provider (134ms)
      ✓ PUT (/) should be able to update a community info (45ms)

  Documents API (/api/v1/documents)
    As anonymous user
      ✓ GET (/) should not be able to list documents
      ✓ POST (/) should not be able to create a document
      ✓ GET (/:id) An anonymous user should not be able to get a unpublished document
      ✓ GET (/:id) An anonymous user should able to get a published document
    As an "Accountable" user (Group member: Accountable - Role: accountable)
      ✓ should get access_token from auth provider (196ms)
      ✓ POST (/) it should be able to create a document (62ms)
      ✓ POST (/) should fail because the permission -documentLimit- is set (Test is limited to 10 tries)
      ✓ GET (/) The author should be able to list its own documents, drafts or published
      ✓ GET (/:id) it should be to retrieve a document
      ✓ PUT (/:id) it should be to update a document (Ex: publishing the document)
      ✓ DELETE (/:id) it should be to delete a document

  Document DB-APIs
    ✓ Document.create() should create a document (54ms)
    ✓ Document.get() should get a document
    ✓ Document.list() should list all documents
    ✓ Document.update() should modify a document
    ✓ Document.remove() should remove a document


  25 passing (6s)
```

### CODE REVIEW

**Comment by guillecro** [comment]
Updated after having a secret conversation with @franciclo

File: db-api/documentTypeVersion.js [hunk 1]
```diff
@@ -11,10 +11,10 @@ exports.get = function get (id) {
 
 exports.getVersion = function getVersion (version) {
   if (version < 0) {
-    throw APIError('Invalid version number')
+    throw errors.ErrInvalidDataAPIError('Invalid version number')
   }
-  return DocumentTypeVersion.findOne({}).then((documentType) => {
-    if (!documentType) throw ErrNotFound(`DocumentType version not found`)
+  return DocumentTypeVersion.findOne({}).lean().then((documentType) => {
```
Comment (franciclo) [inline db-api/documentTypeVersion.js]: que earlyoptimizer que sosss

File: services/auth.js [hunk 2]
```diff
@@ -4,4 +4,49 @@ const mongoStore = require('./sessions')
...
+
+const getPermissions = (req) => {
+  if (req.kauth && req.kauth.grant) {
+    return req.kauth.grant.access_token.content.permissions
+  } else {
+    return null
+  }
+}
+
+module.exports = {
```
Comment (franciclo) [inline services/auth.js:46]: good layer :+1:

**Review by franciclo — APPROVED** [review]
:+1:

**Pull request merged**